src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,382 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.dao.ItemDao;
+import mate.academy.internetshop.dao.OrderDao;
+import mate.academy.internetshop.dao.UserDao;
+import mate.academy.internetshop.exceptions.AuthenticationException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Order;
+import mate.academy.internetshop.model.Role;
+import mate.academy.internetshop.model.User;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.log4j.Logger;
+
+@Dao
+public class UserDaoJdbcImpl extends AbstractDao<User> implements UserDao {
+    private static Logger logger = Logger.getLogger(UserDaoJdbcImpl.class);
+    @Inject
+    private static OrderDao orderDao;
+    @Inject
+    private static ItemDao itemDao;
+    @Inject
+    private static BucketDao bucketDao;
+
+    public UserDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public User add(User user) {
```

Comment:
too long method
your method can't be very large, try to divide it into different parts